src/db/connection.rs

```diff
@@ -0,0 +1,102 @@
+use crate::db::sql_types::*;
+use diesel::pg::PgConnection;
+use rocket::fairing::Fairing;
+use rocket_contrib::databases::r2d2::{Pool, PooledConnection};
+use rocket_contrib::databases::Poolable;
+
+type Connection = PooledConnection<<PgConnection as Poolable>::Manager>;
+
+/// The request guard type.
+pub struct RequestContext {
+    pub conn: Connection,
+}
+
+/// The pool type.
+pub struct ConnectionPool(Pool<<PgConnection as Poolable>::Manager>);
+
+impl RequestContext {
+    /// Create a RequestContext Struct
+    pub fn from_connection(conn: Connection) -> Self {
+        Self { conn }
+    }
+
+    /// Get a DatabaseContext from this struct that can be used to connect
+    /// to the db
+    pub fn database(&self, action: ActionType, modifier: ActionModifier) -> DatabaseContext {
```

Comment:
I'd prefer something like `database_ctx` or `database_context`

Reply:
Because this is *not* the raw DB conn

Reply:
ok fixed!